components/ReferenceManager/references/reference_table/ReferencesTable.tsx

```diff
@@ -246,6 +273,61 @@ export default function ReferencesTable({
       }
     : null;
 
+  const handleSingleClick = (params, event, _details): void => {
+    // Ignore checkbox click
+    if (event.target.matches('input[type="checkbox"], .action-button-class')) {
+      event.stopPropagation();
+      return;
+    }
+
+    if (event.metaKey) {
+      // If meta key is pressed, add or remove from the selection
+      if (selectedRows.includes(params.id)) {
+        handleRowSelection(selectedRows.filter((id) => id !== params.id));
+      } else {
+        handleRowSelection([...selectedRows, params.id]);
+      }
+    } else {
+      // If meta key is not pressed, replace the selection
+      // Without timeout, double click will not be reached since the state update will interrupt it.
+      setTimeout(() => {
```

Comment:
I always get nervous about putting setTimeout's for rendering purposes. Some devices may take longer than 150ms and then it won't work and we have phantom bugs. 

Is there anyway to build without this?

Reply:
Yeah, I understand the nervousness around setTimeout. It will have no impact on rendering. The reason it is needed   is  because we are  using both single  and double click events.  The setTimeout  is recommendation from various sources.  I tried to do without it  -  and it is possible, but the UX suffers a bit